internal/keypool/provider_test.go

```
@@ -91,6 +253,445 @@ func TestProviderStop(t *testing.T) {
+	}
+	require.NoError(t, db.Create(apiKey).Error)
+
+	deleteStarted := make(chan struct{}, 1)
+	callbackName := "test:remove-keys-lock-order"
+	require.NoError(t, db.Callback().Delete().Before("gorm:delete").Register(callbackName, func(_ *gorm.DB) {
+		select {
+		case deleteStarted <- struct{}{}:
+		default:
+		}
+	}))
+	defer db.Callback().Delete().Remove(callbackName)
```

golangci-lint (2.12.2)

[error] Error return value of `(*gorm.io/gorm.processor).Remove` is not checked

(errcheck)